tests/common/mod.rs

```diff
@@ -94,24 +71,54 @@ fn url_from_port(port: String) -> Result<String> {
 }
 
 impl TestConfig {
-    async fn new() -> Result<Self> {
-        let mut load_generator_url =
-            url_from_port(dotenvy::var("ESPRESSO_SUBMIT_TRANSACTIONS_PRIVATE_PORT")?)?;
+    pub async fn new() -> Result<Self> {
+        // We need to know sequencer version we are testing in order
+        // to test against the correct services. It's tempting to try
+        // and proceed against any working services, but we run the
+        // risk of testing against the wrong ones. For example, legacy
+        // builder may be alive during marketplace test.
+        let sequencer_version: u8 = dotenvy::var("INTEGRATION_TEST_SEQUENCER_VERSION")?
+            .parse()
+            .unwrap();
 
-        if dotenvy::var("MARKETPLACE_SMOKE_TEST")
-            .map_err(|e| anyhow!(e))
-            .and_then(|var| var.parse::<bool>().map_err(|e| anyhow!(e)))
-            .ok()
-            .is_some_and(|val| val)
-        {
-            load_generator_url = url_from_port(dotenvy::var(
+        // Varies between v0 and v3.
+        let load_generator_url = if sequencer_version >= 03 {
+            url_from_port(dotenvy::var(
                 "ESPRESSO_SUBMIT_TRANSACTIONS_PRIVATE_FALLBACK_PORT",
-            )?)?;
-        }
+            )?)?
+        } else {
+            url_from_port(dotenvy::var("ESPRESSO_SUBMIT_TRANSACTIONS_PRIVATE_PORT")?)?
+        };
+
+        let builder_url = if sequencer_version >= 03 {
+            let url = url_from_port(dotenvy::var("ESPRESSO_RESERVE_BUILDER_SERVER_PORT")?)?;
```

Comment:
should we use fallback server port here as the load generator has fallback port?

Reply:
Sounds correct.

Reply:
eb3cd72eea26ab5f2218fdb53f68f6b24ff882a7

Reply:
e42ba141277096d99e36cc6e51e8cf48183c9b20

Reply:
So that actually breaks it: https://github.com/EspressoSystems/espresso-sequencer/pull/2086/commits/e42ba141277096d99e36cc6e51e8cf48183c9b20